### PR TITLE
Fix #759

### LINKED ISSUE #759: Collect compiler messages after refchecks
Following up on https://github.com/scalameta/scalameta/pull/749

The semanticdb does still not contain `-deprecation` warnings because they are emitted during refchecks.

### PR DESCRIPTION
**Pull request by olafurpg**
Based on #1076 to align updates on expect tests.

### CODE REVIEW

File: scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/ScalahostPipeline.scala [hunk 1]
```diff
@@ -20,6 +20,48 @@ trait SemanticdbPipeline extends DatabaseOps { self: SemanticdbPlugin =>
...
+            val messages = unit.reportedMessages
+            val mminidb = m.Database(List(mattrs.copy(messages = messages)))
+            mminidb.save(scalametaTargetroot, config.sourceroot)
+          }
+        } catch handleError(unit)
+      }
+    }
+  }
+
   object SemanticdbComponent extends PluginComponent {
```
Comment (xeno-by) [inline scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/ScalahostPipeline.scala]: How about renaming this to `semanticdb-compute`?

File: scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/AttributesOps.scala [hunk 2]
```diff
@@ -40,6 +39,7 @@ trait AttributesOps { self: DatabaseOps =>
         val inferredImplicitConv = mutable.Set.empty[g.Tree] // synthesized implicit conversions
         val todo = mutable.Set[m.Name]() // names to map to global trees
         val mstarts = mutable.Map[Int, m.Name]() // start offset -> tree
+        unit.body.updateAttachment(mstarts) // used in MessagesOps
```
Comment (xeno-by) [inline scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/AttributesOps.scala]: I suggest using something more debuggable than .getClass as attachment keys, e.g. `unit.body.appendMetadata("semanticdbMstarts", mstarts)`.

Comment (olafurpg) [inline scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/AttributesOps.scala]: Done.

File: scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/ScalahostPipeline.scala (hunk 1 above)
Comment (olafurpg) [inline scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/ScalahostPipeline.scala]: Done.

**Comment by olafurpg** [comment]
Thanks for the review! I rebased this PR on top of master and removed the commits from #1076

**Pull request merged**